sympy/core/relational.py

```diff
@@ -1138,3 +1049,204 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+@dispatch(Expr, Expr)
+def _eval_is_le(lhs, rhs):
+    return None
+
+
+@dispatch(Expr, Expr)
+def _eval_is_ge(lhs, rhs):
+    return None
+
+
+@dispatch(Basic, Basic)
+def _eval_is_eq(lhs, rhs):
+    return None
+
+
+@dispatch(Tuple, Expr)
+def _eval_is_eq(lhs, rhs):# noqa:F811
+    return False
+
+
+@dispatch(Tuple, AppliedUndef)
+def _eval_is_eq(lhs, rhs):# noqa:F811
+    return None
+
+
+@dispatch(Tuple, Symbol)
+def _eval_is_eq(lhs, rhs):# noqa:F811
+    return None
+
+
+@dispatch(Tuple, Tuple)
+def _eval_is_eq(lhs, rhs): # noqa:F811
+    if len(lhs) != len(rhs):
+        return False
+
+    return fuzzy_and(fuzzy_bool(is_eq(s, o)) for s, o in zip(lhs, rhs))
+
+
+def is_lt(lhs, rhs):
+    return fuzzy_not(is_ge(lhs, rhs))
+
+
+def is_gt(lhs, rhs):
+    return fuzzy_not(is_le(lhs, rhs))
+
+
+def is_le(lhs, rhs):
+    return is_ge(rhs, lhs)
+
+
+def is_ge(lhs, rhs):
+    if not (isinstance(lhs, Expr) and isinstance(rhs, Expr)):
+        raise TypeError("Can only compare inequalities with Expr")
+
+    retval = _eval_is_ge(lhs, rhs)
+
+    if retval is not None:
+        return retval
+    else:
+        n2 = _n2(lhs, rhs)
+        if n2 is not None:
+            # use float comparison for infinity.
+            # otherwise get stuck in infinite recursion
+            if n2 in (S.Infinity, S.NegativeInfinity):
+                n2 = float(n2)
+            return _sympify(n2 >= 0)
+        if lhs.is_extended_real and rhs.is_extended_real:
+            if (lhs.is_infinite and lhs.is_extended_positive) \
+                     or (rhs.is_infinite and rhs.is_extended_negative):
+                return True
+            diff = lhs - rhs
+            if diff is not S.NaN:
+                rv = diff.is_extended_nonnegative
+                if rv is not None:
+                   return rv
+
+
+def is_neq(lhs, rhs):
+    return fuzzy_not(is_eq(lhs, rhs))
+
+
+def is_eq(lhs, rhs):
```

Comment:
These top-level functions `is_eq`, `is_ne`, etc should have docstrings with examples. They don't need to be added to the docs but an explanation and a clear tested demonstration would be good. There could be one docstring for `is_eq` and one for `is_ge` and the others can refer to them for more details e.g.:
```
def is_neq(lhs, rhs):
    """Fuzzy bool for lhs does not equal rhs.

    See the docstring for is_eq for more
    """
```

I think it would be good for the docstring for `is_eq` to explain (with an example) that `_eval_is_eq` is used for both `is_eq` and `is_ne`. Likewise the `is_ge` docstring should explain how to overload `_eval_is_ge` and that it covers all of `is_le` etc.

Reply:
I've added these top level functions, but I'm not sure if there is a way to include an example with overload _eval_is_ge etc. when doing the docstring

Reply:
You could make up a class, something like
```python
def is_ge(lhs, rhs):
    """
    >>> class Power(Expr):
    ...     def __new__(cls, base, exp):
    ...         return Basic.__new__(_sympify(base), _sympify(exp))
    >>> @dispatch(Power, Power):
    ... def _eval_is_ge(lhs, rhs):
    ...     if fuzzy_and(is_ge(lhs.args[0], rhs.args[0]), is_ge(lhs.args[1], rhs.args[1]))
    ...         return True
    >>> Power(2, 3) < Power(3, 6)
    True
    """
```

Reply:
There should be a more involved example in the tests, including tests that `_eval_Eq` still works.

Reply:
Should the _eval_Eq portion be in the unit tests, or the doctests or both?

Reply:
That should be the unit tests. The doctests should demonstrate the code we want users to use.